code/modules/mob/living/carbon/shock.dm

```diff
@@ -1,47 +1,150 @@
+#define SOFTCRIT_TRAUMATIC_SHOCK	50
+#define HARDCRIT_TRAUMATIC_SHOCK	100
+#define SHOCK_STAGE_BUFFER			50
+
 /mob/living/var/traumatic_shock = 0
 /mob/living/carbon/var/shock_stage = 0
 
 // proc to find out in how much pain the mob is at the moment
 /mob/living/carbon/proc/updateshock()
-	if (species && (species.flags & NO_PAIN))
-		src.traumatic_shock = 0
+	if(species && (species.flags & NO_PAIN))
+		traumatic_shock = 0
 		return 0
 
-	src.traumatic_shock = 			\
-	1	* src.getOxyLoss() + 		\
-	0.5	* src.getToxLoss() + 		\
-	1	* src.getPainFromDam() +	\
-	1.7	* src.getCloneLoss() + 		\
-	2	* src.halloss + 			\
-	-1	* src.analgesic
+	traumatic_shock = get_constant_pain() + get_dynamic_pain() - get_painkiller()
+
+	if(slurring)
+		traumatic_shock -= 10
 
-	if(src.slurring)
-		src.traumatic_shock -= 20
+	if(traumatic_shock < 0)
+		traumatic_shock = 0
 
-	if(src.traumatic_shock < 0)
-		src.traumatic_shock = 0
+	return traumatic_shock
 
-	return src.traumatic_shock
+/mob/living/carbon/var/last_tick_pain
 
-/mob/living/carbon/proc/getPainFromDam()
-	return src.getFireLoss() + src.getBruteLoss()
+/mob/living/carbon/proc/get_constant_pain()
+	var/hard_crit_threshold = HARDCRIT_TRAUMATIC_SHOCK + min(stats.getStat(STAT_TGH), 100)
```

Comment:
Are we sure all `/mob/living/carbon` got stats?

Reply:
All mobs have stats I believe

Reply:
Yes, all mob subtypes have stats.